Blender is broken on 32bit

https://build.haiku-os.org/buildmaster/master/x86_gcc2/logviewer.html?buildruns/870/builds/1490.log

Think this is the root cause: undefined reference to `___tls_get_addr'

--- a/media-gfx/blender/blender-2.79b.recipe
+++ b/media-gfx/blender/blender-2.79b.recipe
@@ -8,9 +8,9 @@ application and write specialized tools; often these are included in Blender’s
 future releases. Blender is well suited to individuals and small studios who \
 benefit from its unified pipeline and responsive development process."
 HOMEPAGE="https://www.blender.org/"
-COPYRIGHT="2002-2017 Blender Foundation"
+COPYRIGHT="2002-2018 Blender Foundation"
 LICENSE="GNU GPL v2"
-REVISION="21"
+REVISION="22"
 SOURCE_URI="https://download.blender.org/source/blender-$portVersion.tar.gz"
 CHECKSUM_SHA256="4c944c304a49e68ac687ea06f5758204def049b66dc211e1cffa1857716393bc"
 PATCHES="
@@ -22,7 +22,7 @@ ADDITIONAL_FILES="
 	"
 
 ARCHITECTURES="!x86_gcc2 x86_64"
-SECONDARY_ARCHITECTURES="!x86"
+SECONDARY_ARCHITECTURES="x86"
 
 PYTHON_VERSION="3.6"
 
@@ -33,8 +33,8 @@ PROVIDES="
 	"
 REQUIRES="
 	haiku$secondaryArchSuffix
-	numpy${secondaryArchSuffix}_python36
-	requests_python36
+#	numpy${secondaryArchSuffix}_python3
+#	requests_python3
 	lib:libalembic$secondaryArchSuffix
 	lib:libavcodec$secondaryArchSuffix
 	lib:libavdevice$secondaryArchSuffix
@@ -50,13 +50,10 @@ REQUIRES="
 	lib:libfftw3$secondaryArchSuffix
 	lib:libfreetype$secondaryArchSuffix
 	lib:libgcc_s$secondaryArchSuffix
-#	lib:libgflags$secondaryArchSuffix
 	lib:libgl$secondaryArchSuffix
 	lib:libglew$secondaryArchSuffix
-#	lib:libglog$secondaryArchSuffix
 	lib:libglu$secondaryArchSuffix
 	lib:libgomp$secondaryArchSuffix
-#	lib:libgtest$secondaryArchSuffix
 	lib:libhalf_2_4$secondaryArchSuffix
 	lib:libhdf5$secondaryArchSuffix
 	lib:libiconv$secondaryArchSuffix
@@ -64,7 +61,6 @@ REQUIRES="
 	lib:libilmimf_2_4$secondaryArchSuffix
 	lib:libilmthread_2_4$secondaryArchSuffix
 	lib:libimath_2_4$secondaryArchSuffix
-	lib:libjpeg$secondaryArchSuffix
 	lib:libLLVM_9$secondaryArchSuffix
 	lib:liblzma$secondaryArchSuffix
 	lib:liblzo2$secondaryArchSuffix
@@ -72,8 +68,7 @@ REQUIRES="
 	lib:libopencolladabaseutils$secondaryArchSuffix
 	lib:libopencolorio$secondaryArchSuffix
 	lib:libopenimageio$secondaryArchSuffix
-#	lib:libopenjp2$secondaryArchSuffix
-	lib:libopenvdb$secondaryArchSuffix
+#	lib:libopenvdb$secondaryArchSuffix
 	lib:libosdCPU$secondaryArchSuffix
 	lib:libosdGPU$secondaryArchSuffix
 #	lib:liboslcomp$secondaryArchSuffix
@@ -86,6 +81,7 @@ REQUIRES="
 	lib:libtbb$secondaryArchSuffix
 	lib:libtiff$secondaryArchSuffix
 	lib:libtinyxml$secondaryArchSuffix
+	lib:libturbojpeg$secondaryArchSuffix
 	lib:libxml2$secondaryArchSuffix
 	lib:libyaml_cpp$secondaryArchSuffix
 	lib:libz$secondaryArchSuffix
@@ -108,13 +104,10 @@ BUILD_REQUIRES="
 	devel:libexecinfo$secondaryArchSuffix
 	devel:libfftw3$secondaryArchSuffix
 	devel:libfreetype$secondaryArchSuffix
-#	devel:libgflags$secondaryArchSuffix
 	devel:libgl$secondaryArchSuffix
 	devel:libglew$secondaryArchSuffix
-#	devel:libglog$secondaryArchSuffix
 	devel:libglu$secondaryArchSuffix
 	devel:libgomp$secondaryArchSuffix
-#	devel:libgtest$secondaryArchSuffix
 	devel:libhalf_2_4$secondaryArchSuffix
 	devel:libhdf5$secondaryArchSuffix
 	devel:libiconv$secondaryArchSuffix
@@ -122,7 +115,6 @@ BUILD_REQUIRES="
 	devel:libilmimf_2_4$secondaryArchSuffix
 	devel:libilmthread_2_4$secondaryArchSuffix
 	devel:libimath_2_4$secondaryArchSuffix
-	devel:libjpeg$secondaryArchSuffix
 	devel:libllvm_9$secondaryArchSuffix
 	devel:liblzma$secondaryArchSuffix
 	devel:liblzo2$secondaryArchSuffix
@@ -130,8 +122,7 @@ BUILD_REQUIRES="
 	devel:libopencolladabaseutils$secondaryArchSuffix
 	devel:libopencolorio$secondaryArchSuffix
 	devel:libopenimageio$secondaryArchSuffix
-#	devel:libopenjp2$secondaryArchSuffix
-	devel:libopenvdb$secondaryArchSuffix
+#	devel:libopenvdb$secondaryArchSuffix
 	devel:libosdCPU$secondaryArchSuffix
 	devel:libosdGPU$secondaryArchSuffix
 #	devel:liboslcomp$secondaryArchSuffix
@@ -144,14 +135,15 @@ BUILD_REQUIRES="
 	devel:libtbb$secondaryArchSuffix
 	devel:libtiff$secondaryArchSuffix
 	devel:libtinyxml$secondaryArchSuffix
+	devel:libturbojpeg$secondaryArchSuffix
 	devel:libxml2$secondaryArchSuffix
 	devel:libyaml_cpp$secondaryArchSuffix
 	devel:libz$secondaryArchSuffix
 	lib:libavdevice$secondaryArchSuffix # REQUIRED
 	"
 BUILD_PREREQUIRES="
-	numpy${secondaryArchSuffix}_python36
-	requests_python36
+#	numpy${secondaryArchSuffix}_python3
+#	requests_python3
 	cmd:cmake
 	cmd:find
 	cmd:gcc$secondaryArchSuffix
@@ -214,6 +206,8 @@ BUILD()
 		-DWITH_PYTHON_INSTALL=OFF \
 		-DWITH_PYTHON_INSTALL_NUMPY=OFF \
 		-DWITH_PYTHON_INSTALL_REQUESTS=OFF \
+		-DWITH_PYTHON_MODULE=OFF \
+		-DWITH_PYTHON_SAFETY=OFF \
 		-DCMAKE_SKIP_RPATH=ON \
 		-Wno-dev"
 

